### PR TITLE
[Kotlin Migration] HttpFetcher

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/web/HttpFetcher.kt [hunk 1]
```diff
@@ -18,159 +18,146 @@
...
+                InputStreamReader(
+                    response.body!!.byteStream(),
+                    Charset.forName(encoding)
+                )
+            )
+            val stringBuilder = StringBuilder()
+            var line: String?
+            while (reader.readLine().also { line = it } != null) {
+                stringBuilder.append(line)
             }
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/web/HttpFetcher.kt:112]: `@KotlinCleanup` - maybe I need to be taught this is idiomatic, but it's strange to me

File: AnkiDroid/src/main/java/com/ichi2/anki/web/HttpFetcher.kt [hunk 2]
```diff
@@ -18,159 +18,146 @@
...
+            stringBuilder.toString()
+        } catch (e: Exception) {
+            Timber.d(e, "Failed with an exception")
+            "FAILED with exception: " + e.message
         } finally {
-            if (response != null && response.body() != null) {
-                response.body().close();
+            if (response?.body != null) {
+                response.body!!.close()
             }
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/web/HttpFetcher.kt:120]: KotlinCleanup("response?.body?.close()")

File: AnkiDroid/src/main/java/com/ichi2/anki/web/HttpFetcher.kt [hunk 3]
```diff
@@ -18,159 +18,146 @@
...
+            file.absolutePath
+        } catch (e: Exception) {
+            Timber.w(e)
+            "FAILED " + e.message
         } finally {
-            if (response != null && response.body() != null) {
-                response.body().close();
+            if (response?.body != null) {
+                response.body!!.close()
             }
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/web/HttpFetcher.kt:160]: KotlinCleanup(`response?.body?.close()`)

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
few cleanups

**Comment by david-allison-1** [comment]
CI failure - likely a bad test, rerunning

```
com.ichi2.anki.servicemodel.UpgradeVolumeButtonsToBindingsTest > gesture_set_no_conflicts[0: isValid(TestData(affectedPreferenceKey=gestureVolumeUp, keyCode=24, unaffectedPreferenceKey=gestureVolumeDown, binding=com.ichi2.anki.reviewer.MappableBinding@1900db))={1}] FAILED
    java.lang.AssertionError: 
    Expected: iterable over ["preferenceUpgradeVersion", "gestureVolumeUp", "binding_LOOKUP"] in any order
         but: No item matches: "preferenceUpgradeVersion", "gestureVolumeUp", "binding_LOOKUP" in []
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
        at com.ichi2.anki.servicemodel.UpgradeVolumeButtonsToBindingsTest.gesture_set_no_conflicts(UpgradeVolumeButtonsToBindingsTest.kt:97)
```

**Review by david-allison-1 — APPROVED** [review]
LGTM

File: AnkiDroid/src/main/java/com/ichi2/anki/web/HttpFetcher.kt [hunk 4]
```diff
@@ -18,159 +18,150 @@
...
         } else {
-            clientBuilder.addNetworkInterceptor(chain -> chain.proceed(
-                    chain.request()
+            clientBuilder.addNetworkInterceptor(
+                Interceptor { chain: Interceptor.Chain ->
+                    chain.proceed(
+                        chain.request()
                             .newBuilder()
-                            .header("User-Agent", "AnkiDroid-" + VersionUtils.getPkgVersionName())
+                            .header("User-Agent", "AnkiDroid-$pkgVersionName")
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/web/HttpFetcher.kt:79]: kotlin string interpolation is nifty

**Review by mikehardy — APPROVED** [review]

**Pull request merged**